OpenRA.Mods.Common/Traits/BotModules/Squads/States/NavyStates.cs

```diff
@@ -100,7 +101,17 @@ public void Tick(Squad owner)
 			if (!owner.IsValid)
 				return;
 
-			if (!owner.IsTargetValid)
+			// Check if we have an enemy naval yard this tick
```

Comment:
I'm a bit uncomfortable about checking all this every tick, for performance reasons.
I think it would be better if the squad only checked every second (25 ticks) or so.

Reply:
From my understanding of the squad code, this tick only called on the AttackForceInterval. Its called by the fuzzystatemachine which only updates when AttackForceTick <= 0 in squadmanager. So by default its only checking every 75 ticks.